### PR TITLE
Add user suggestion search and self-activity lookup

### PR DESCRIPTION
**Pull request by Nahydrin**
- Adds /activity
  - Can only view your own activity
- Adds /suggestions
  - Can search by a specific member
  - Can search by all members
  - Can filter by one or more tags
  - Can filter by a phrase in the title

### CODE REVIEW

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 1]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
...
+
+        if (suggestionForumIds == null || suggestionForumIds.length == 0) {
+            event.reply("No " + (isAlpha ? "alpha " : "") + "suggestion forums are setup in the config.").setEphemeral(true).queue();
+            return;
+        }
+
+        List<Suggestion> suggestions = getSuggestions(suggestionForumIds, searchMember, tags, title, isAlpha);
+
+        if (suggestions.isEmpty()) {
+            event.reply("You have no suggestions matched with tags " + (tags != null ? ("`" + tags + "`") : "*ANY*") + " or title containing " + (title != null ? ("`" + title + "`") : "*ANYTHING*") + ".").setEphemeral(true).queue();
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:245]: Can you rephrase this message 

Maybe something like this to keep it simple:
Found no suggestions matching the specified filters!

Alternatively just make the formatting of the message better when there's no tags/title etc

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 2]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
         Instant date = Instant.now().plus(duration);
         return Date.from(date);
     }
+
+    @JDASlashCommand(name = "suggestions", subcommand = "by-member", description = "View user suggestions.")
+    public void mySuggestions(
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:222]: Rename to viewOwnSuggestions

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 3]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
...
+        if (suggestions.isEmpty()) {
+            event.reply("You have no suggestions matched with tags " + (tags != null ? ("`" + tags + "`") : "*ANY*") + " or title containing " + (title != null ? ("`" + title + "`") : "*ANYTHING*") + ".").setEphemeral(true).queue();
+            return;
+        }
+
+        event.replyEmbeds(buildSuggestionsEmbed(suggestions, tags, title, isAlpha, pageNum).setAuthor(searchMember.getEffectiveName()).build()).setEphemeral(true).queue();
+    }
+
+    @JDASlashCommand(name = "suggestions", subcommand = "by-everyone", description = "View all suggestions.")
+    public void allSuggestions(
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:253]: Rename to viewAllSuggestions

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 4]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
...
+        if (suggestions.isEmpty()) {
+            event.reply("You have no suggestions matched with tags " + (tags != null ? ("`" + tags + "`") : "*ANY*") + " or title containing " + (title != null ? ("`" + title + "`") : "*ANYTHING*") + ".").setEphemeral(true).queue();
+            return;
+        }
+
+        event.replyEmbeds(buildSuggestionsEmbed(suggestions, tags, title, isAlpha, pageNum).build()).setEphemeral(true).queue();
+    }
+
+    @JDASlashCommand(name = "activity", description = "View your recent activity.")
+    public void myActivity(GuildSlashEvent event) {
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:282]: Rename to viewOwnActivity

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 5]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
...
+                    .stream()
+                    .anyMatch(forumTag -> forumTag.getName().equalsIgnoreCase(tag))
+                ))
+                .filter(thread -> title == null || thread.getName()
+                    .toLowerCase()
+                    .contains(title.toLowerCase())
+                )
+                .map(thread -> new Suggestion(thread, alpha))
+            )
+            .toList();
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:317]: This makes me cry

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 6]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
...
+            .setDescription(
+                (tags != null ? "- Filtered by tags: `" + tags + "`\n" : "") + (title != null ? "- Filtered by title: `" + title + "`" : "")
+            )
+            .addField(
+                "Total",
+                String.valueOf((int) total),
+                true
+            )
+            .addField(
+                ("<:creative:" + emojiConfig.getGreenlitEmojiId() + ">"),
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:356]: Can we change this to use `jda.getEmojiById(id)` instead? I think it should still show the actually emoji

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 7]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
...
+            .addBlankField(true)
+            .addField(
+                "Links",
+                fieldData.stream()
+                    .map(list -> list.get(0))
+                    .collect(Collectors.joining("\n")),
+                true
+            )
+            .addField(
+                ("<:agree:" + emojiConfig.getAgreeEmojiId() + ">"),
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:369]: See above

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java [hunk 8]
```diff
@@ -198,4 +217,253 @@ public static Date parseCustomFormat(String time) throws DateTimeParseException
...
+                fieldData.stream()
+                    .map(list -> list.get(1))
+                    .collect(Collectors.joining("\n")),
+                true
+            )
+            .addField(
+                ("<:disagree:" + emojiConfig.getDisagreeEmojiId() + ">"),
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/UserCommands.java:376]: See above

**Review by Aerhhh — CHANGES_REQUESTED** [review]

File: src/main/java/net/hypixel/nerdbot/command/UserCommands.java (hunk 5 above)
Comment (Nahydrin) [reply src/main/java/net/hypixel/nerdbot/command/UserCommands.java:317]: Good
